docs/sync/azure_pipelines.md

```diff
@@ -0,0 +1,316 @@
+```
+
+Only the last tag reference, by SHA256 digest, is guaranteed to not have the underlying image it points to change.
+
+[resource_container]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/resources-containers-container
+[step_target]: https://learn.microsoft.com/azure/devops/pipelines/process/tasks#step-target
+[yaml_job_container]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/jobs-job-container
+[yaml_resources]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/resources
+
+### Script arguments
+
+The arguments to the script step are the way to exert control over the execution of the Phylum analysis.
+The entry here will run as a script in the `phylum-ci` based container job.
+See the [YAML schema steps.script definition][yaml_script] and [container job][container_job] documentation for
+more detail.
+
+The `phylum-ci` script entry point is expected to be called. It has a number of arguments that are all optional
+and defaulted to secure values. To view the arguments, their description, and default values,
+run the script with `--help` output as specified in the [Usage section of the top-level README.md][usage] or
+view the [script options output][script_options] for the latest release.
+
+[yaml_script]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/steps-script
+[script_options]: https://github.com/phylum-dev/phylum-ci/blob/main/docs/script_options.md
+
+```yaml
+      # NOTE: These are examples. Only one script entry line for `phylum-ci` is expected.
+
+      # Use the defaults for all the arguments.
+      # The default behavior is to only analyze newly added dependencies against
+      # the risk domain threshold levels set at the Phylum project level.
+      - script: phylum-ci
+
+      # Consider all dependencies in analysis results instead of just the newly added ones.
+      # The default is to only analyze newly added dependencies, which can be useful for
+      # existing code bases that may not meet established project risk thresholds yet,
+      # but don't want to make things worse. Specifying `--all-deps` can be useful for
+      # casting the widest net for strict adherence to Quality Assurance (QA) standards.
+      - script: phylum-ci --all-deps
+
+      # Some lockfile types (e.g., Python/pip `requirements.txt`) are ambiguous in that
+      # they can be named differently and may or may not contain strict dependencies.
+      # In these cases, it is best to specify an explicit lockfile path.
+      - script: phylum-ci --lockfile requirements-prod.txt
+
+      # Thresholds for the five risk domains may be set at the Phylum project level.
+      # They can be set differently for CI environments to "fail the build."
+      # NOTE: The shortened form is used here for brevity, but the long form might be more
+      # descriptive for future readers. For instance, `--vul-threshold` instead of `-u`.
```

Comment:
Handling this in a separate issue is fine. But this makes it even more clear that we have an issue worth dealing with here. Because at the moment we are making a recommendation while also considering it too complicated for us to implement ourselves. If it's truly that complicated, we should remove the note. But I'm hopeful that we can get it figured out.

Reply:
#128 was created to track this. Resolving this conversation now...